decentraland/unity-explorer
# Optimize Avatar Attach Setup / UpdateAvatarAttachTransform (#1396)

The SetupAvatarAttach method has been moved to a new system. The new system executes before AvatarAttachedHandlerSystem and uses Throttling. Auxiliary methods have been moved to a static class, as it is shared among both systems.

## What does this PR change?

It allows using Throttling on SetupAvatarAttach.

## How to test the changes?

1. Launch the explorer using a local scene (see the example below).
2. The entity should attach properly to the avatar from the very first moment.
3. Verify that the scenes in Goerli which include attaching objects to the avatar still work.

```
Example code for a local scene:
export function main() {
    const entity = engine.addEntity()
    Transform.create(entity)
    MeshRenderer.setBox(entity)   
    AvatarAttach.create(entity, { anchorPointId: AvatarAnchorPointType.AAPT_LEFT_LEG});
}
```
